tests/bot/cogs/moderation/test_utils.py

```diff
@@ -0,0 +1,411 @@
+            },
+            {
+                "args": (self.user, "Test title", "Example content", Icons.user_update),
+                "icon": Icons.user_update,
+                "send_result": False
+            }
+        ]
+
+        for case in test_cases:
+            args = case["args"]
+            send = case["send_result"]
+
+            expected = Embed(
+                description="Example content",
+                colour=Colours.soft_green
+            ).set_author(
+                name="Test title",
+                icon_url=case["icon"]
+            )
+
+            with self.subTest(args=args, expected=expected):
+                send_private_embed_mock.reset_mock()
+
+                send_private_embed_mock.return_value = send
+
+                result = await utils.notify_pardon(*args)
+                self.assertEqual(send, result)
+
+                embed = send_private_embed_mock.call_args[0][1]
+                self.assertEqual(embed.to_dict(), expected.to_dict())
+
+                send_private_embed_mock.assert_awaited_once_with(args[0], embed)
+
+    @patch("bot.cogs.moderation.utils.log")
+    async def test_post_user(self, log_mock):
+        """Should POST a new user and return the response if successful or otherwise send an error message."""
+        user = MockUser(discriminator=5678, id=1234, name="Test user")
+        some_mock = MagicMock(discriminator=3333)
+        test_cases = [
+            {
+                "user": user,
+                "post_result": "bar",
+                "raise_error": None,
+                "payload": {
+                    "discriminator": 5678,
+                    "id": self.user.id,
+                    "in_guild": False,
+                    "name": "Test user",
+                    "roles": []
+                }
+            },
+            {
+                "user": self.member,
+                "post_result": "foo",
+                "raise_error": ResponseCodeError(MagicMock(status=400), "foo"),
+                "payload": {
+                    "discriminator": 0,
+                    "id": self.member.id,
+                    "in_guild": False,
+                    "name": "Name unknown",
+                    "roles": []
+                }
+            },
+            {
+                "user": some_mock,
+                "post_result": "bar",
+                "raise_error": None,
+                "payload": {
+                    "discriminator": some_mock.discriminator,
+                    "id": some_mock.id,
+                    "in_guild": False,
+                    "name": some_mock.name,
+                    "roles": []
+                }
+            }
+        ]
+
+        for case in test_cases:
+            test_user = case["user"]
+            expected = case["post_result"]
+            error = case["raise_error"]
+            payload = case["payload"]
+
+            with self.subTest(user=test_user, result=expected, error=error, payload=payload):
```

Comment:
I'd strongly prefer if the names you're assigning to matched the keys. This creates even more indirection between the definitions above and the code below. You're using 3 different names (key, local variable, `subTest` kwarg) for the same thing.

Reply:
Synced all names.